### ISSUE TITLE
XLNet ONNX model giving error: "Attempting to broadcast an axis by a dimension other than 1"

### ISSUE THREAD

**Issue by singhn27** [microsoft/onnxruntime#6130]
**Describe the bug**
Attempting to run inference in ONNXRuntime with an XLNetForSequenceClassification model trained with Huggingface Transformers and converted to ONNX using their provided `convert_graph_to_onnx.py` script, I get the following error:

```RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running Add node. Name:'Add_26' Status Message: /onnxruntime_src/onnxruntime/core/providers/cpu/math/element_wise_ops.h:361 void onnxruntime::BroadcastIterator::Init(int64_t, int64_t) axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 5 by 6```

**Urgency**
Medium-term project-related deadline

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Amazon Linux AMI 2018.03
- ONNX Runtime installed from (source or binary): binary
- ONNX Runtime version: 1.4.0, 1.6.0 (tried both)
- Python version: 3.7.6
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 11.0
- GPU model and memory: Tesla V100, 16 GB

**To Reproduce**
- Describe steps/code to reproduce the behavior.
- Attach the ONNX model to the issue (where applicable) to expedite investigation.

1. Trained HuggingFace Transformers model XLNetForSequenceClassification on custom dataset with PyTorch backend.
2. Used provided convert_graph_to_onnx.py script to convert model (from saved checkpoint) to ONNX format.
3. Loaded the model with ONNXRuntime
4. When feeding in int64 numpy arrays input_ids and attention_masks, the model returns the following error except when both inputs have shape (x, 1) or (x, 6). There is nothing in the configuration of the model or the structure of the training data from my end that would require shape (x, 1) or (x, 6). Here is sample code for a test input of sequence length 5.

```import onnxruntime as rt
import numpy as np

sess = rt.InferenceSession('xlnet.onnx')
input_ids = np.array([[1,30,5,4,2]]).astype(np.int64)
attention_mask = np.array([[1,1,1,1,1]]).astype(np.int64)
pred_onx = sess.run(None, {'input_ids': input_ids, 'attention_mask': attention_mask})[0]
```

**Expected behavior**
The expected behavior is for the model to return predictions successfully (i.e. probabilities for all classes, in this case 77).

**Screenshots**

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

Particular model: XLNet

Note, others are facing a similar problem. I am also linking two other places where I have shared this bug.

https://github.com/huggingface/transformers/issues/8542
https://github.com/huggingface/transformers/issues/9044
https://discuss.huggingface.co/t/xlnet-onnx-model-giving-error-attempting-to-broadcast-an-axis-by-a-dimension-other-than-1/2784

**Comment by wangyems** [comment]
XLNet does have issue with ORT when running on GPU. It seems to me that your issue does not related with that. Could you first try  inferencing the pytorch model with the same input size as inputs you used in ORT?

**Comment by singhn27** [comment]
@wangyems I tried inference with the PyTorch model using the same input size as used in ORT and it worked fine without any errors. Please let me know of next steps on how we can diagnose and fix this issue. Thank you.

**Comment by wangyems** [comment]
If the input size for your model in fixed, you could just export the onnx model using same sized input when feeding pytorch model.
If the input size is dynamic, you could refer https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/transformers/onnx_exporter.py#L346 which specify dynamic axis before export.

**Comment by wangyems** [comment]
Closing the issue. Please reopen it if there's still questions
